ethereum/p2p/src/main/java/tech/pegasys/pantheon/ethereum/p2p/network/DefaultP2PNetwork.java

```diff
@@ -595,38 +590,36 @@ private synchronized void checkCurrentConnections() {
         .getPeerConnections()
         .forEach(
             peerConnection -> {
-              if (!isPeerConnectionAllowed(peerConnection)) {
+              if (!isPeerAllowed(peerConnection)) {
                 peerConnection.disconnect(DisconnectReason.REQUESTED);
               }
             });
   }
 
-  private boolean isPeerConnectionAllowed(final PeerConnection peerConnection) {
-    if (peerBlacklist.contains(peerConnection)) {
-      return false;
-    }
-
-    LOG.trace(
-        "Checking if connection with peer {} is permitted",
-        peerConnection.getPeerInfo().getNodeId());
-
-    return nodePermissioningController
-        .map(
-            c -> {
-              final EnodeURL localPeerEnodeURL = getLocalEnode().orElse(buildSelfEnodeURL());
-              final EnodeURL remotePeerEnodeURL = peerConnection.getRemoteEnode();
-              return c.isPermitted(localPeerEnodeURL, remotePeerEnodeURL);
-            })
-        .orElse(true);
+  private boolean isPeerAllowed(final PeerConnection conn) {
+    return isPeerAllowed(conn.getRemoteEnode());
   }
 
   private boolean isPeerAllowed(final Peer peer) {
-    if (peerBlacklist.contains(peer)) {
+    return isPeerAllowed(peer.getEnodeURL());
+  }
+
+  private boolean isPeerAllowed(final EnodeURL enode) {
+    if (peerBlacklist.contains(enode.getNodeId())) {
+      return false;
+    }
+    if (enode.getNodeId().equals(ourPeerInfo.getNodeId())) {
+      // Peer matches our node id
       return false;
     }
 
+    Optional<EnodeURL> maybeEnode = getLocalEnode();
+    if (!maybeEnode.isPresent()) {
```

Comment:
Be restrictive around permissions until we have a valid representation of our local enode.